### DISCUSSION TITLE
Is their any way to support a BigDecimal column?

### DISCUSSION THREAD

**Discussion by hmf** [jtablesaw/tablesaw#1154] [Q&A] [answered]
Ideally we should handle currency as a `BigDecimal`. Is their any way one can "hack" JTablesaw to have such a  type column? Could it be as simple as extending the NumberColumn?

TIA

**Comment by lwhite1** [comment, marked as answer]
That would be the way to go. The issues i would expect is that a lot of the
math logic relies on + - / * and almost everything (including the summary
methods like "sum" and "mean") convert the data to double so there might be
a fair amount of work there.

(I really wish the java maintainers would come out of the 80s and provide
operator overloading.)

**Reply by hmf** [reply]
@lwhite1 Thanks for the feedback. I won't try to tackle this unless my hack doesn't work.

**Comment by perNyfelt** [comment]
I did this quite successfully (https://github.com/Alipsa/data-utils/blob/master/src/main/java/tech/tablesaw/api/BigDecimalColumn.java) but i am using it from groovy where there is automatic conversion of operators to the equivalent BigDecimal method. I will check how it behaves in a pure Java environment and submit a PR if positive.

**Reply by bhanu-addepar** [reply]
@perNyfelt Have you checked it for the pure Java env?
